mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageManager.java

```diff
@@ -393,6 +394,7 @@ public ComposedMessageId appendMessage(InputStream msgIn, Date internalDate, fin
                     // the file now
                     // via the TeeInputStream
                 }
+                bufferedOut.close();
```

Comment:
why?

Reply:
Otherwise, the message does not get fully copied in the Cassandra implementation. Adding this 'close' solves the issue.

Reply:
maybe use flush() instead ?